lib/arrays.md

```diff
@@ -5,44 +5,58 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is an Array?
 
-Replace me with your response...
+An array is used to organize information into an ordered list. 
+The list can be made up of strings, integers, floats, booleans, or any other data type. 
 
 # What are some examples of information that would be Arrays as opposed to some other data type?
 
-Replace me with your response...
+In Ruby, an array is denoted by square brackets [ ]. Inside the brackets you can create a list of elements separated by commas. 
+The brackets define that it would an array.
 
 # What is one way, using Ruby, to retrieve the 6th element in an Array? How about the 8th element? What happens if you try to retrieve the value of the _9999th_ element (or some element that doesn't exist in the array)?
 
-Replace me with your response...
+Each element in an array can be accessed via an index. 
+For example [ 1, 2, 3, 4, 5][0] would get us the '1' from the list because the values start at '0'.
+if you try to retrieve the 9999th number it would come back as 'nil' or nonexistant.
 
 # The previous question asks about finding, for example, the 6th element in an Array. Is it possible to find the **-6th** (Notice the negative symbol!) element in an Array? What does that even mean?
 
-Replace me with your response...
+it is possible to retrieve the -6th element. If I did [ 1, 2, 3, 4, 5, 6, 7, 8, 9][-6] it would give me 4. 
+It starts at the end of my index and counts backwards 6.
 
 # How would you perform an operation on every element inside an Array?
 
-Replace me with your response...
+you would use something similar to Array.[](...) [or] Array[...] [or] [...] depending on what it is you wanted to accomplish. Like Array.index or Array.length.
```

Comment:
Let's say you have the following code:

```ruby
friends = ["Dan", "Mindy", "Beily"]
```

And you want to send a letter to each friend (Pretend we have a handy Ruby function called `send_letter`).

Your answer suggests that we could perform the operation on each friend like so:

```ruby
send_letter(friends[0])
send_letter(friends[1])
send_letter(friends[2])
```

That is correct. But there are two challenges with this kind of approach:

1. When we write our code, we need to know ahead of time how many elements are in the array. Otherwise, we won't know how much times to type our `send_letter(friends[n])`.
2. We have to repeat ourselves! Coders hate repeating themselves :) The repetition here is that we needed to type `send_letter...` once for each friend.

Ruby–and most programming languages–have a standard way of accomplishing our goal without compromising on the two challenges above. Have you found out what that is? If not, please do some more research and get back to me. And of course, don't hesitate to ask for guidance.